Mathlib/Data/MvPolynomial/Variables.lean

```diff
@@ -824,15 +824,16 @@ theorem hom_congr_vars {f₁ f₂ : MvPolynomial σ R →+* S} {p₁ p₂ : MvPo
 
 theorem exists_rename_eq_of_vars_subset_range (p : MvPolynomial σ R) (f : τ → σ) (hfi : Injective f)
     (hf : ↑p.vars ⊆ Set.range f) : ∃ q : MvPolynomial τ R, rename f q = p :=
-  ⟨aeval (fun i : σ => Option.elim' 0 X <| partialInv f i) p, by
-    show (rename f).toRingHom.comp _ p = RingHom.id _ p
-    refine' hom_congr_vars _ _ _
-    · ext1
-      simp [algebraMap_eq]
-    · intro i hip _
-      rcases hf hip with ⟨i, rfl⟩
-      simp [partialInv_left hfi]
-    · rfl⟩
+  ⟨aeval (fun i : σ => Option.elim' 0 X <| partialInv f i) p,
+    by
```

Comment:
Not going to require you to change it back, but an exception to the exception here is that if the surrounding tuple starts on the same line as the hanging `by`, it is okay. The main thing to avoid is that you have a `by` starting an item where the previous sibling ends earlier on the same line and starts somewhere on a previous line, as in that case it is hard to find the parent of the `by` subexpression by following indentation / going backward in the line.